python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py

```diff
@@ -729,20 +729,25 @@ def resolve_instigationStateOrError(
         instigationSelector: Optional[GrapheneInstigationSelector] = None,
         id: Optional[str] = None,
     ):
+        selector = (
+            InstigatorSelector.from_graphql_input(instigationSelector)
+            if instigationSelector
+            else None
+        )
         if id:
+            # prefer loading by id if possible for performance, falls back to selector if provided
             return get_instigation_state_by_id(
                 graphene_info,
                 CompoundID.from_string(id),
+                selector,
             )
-        elif instigationSelector:
+        elif selector:
             return get_instigator_state_by_selector(
                 graphene_info,
-                InstigatorSelector.from_graphql_input(instigationSelector),
+                selector,
             )
```

Comment:
Should we just require both the ID and the selector ?

Reply:
yea could sort of revert back closer to how this used to be with id as an optional accelerant argument

Reply:
i like that framing